### PR TITLE
LndRestWallet: catch http errors

### PR DESCRIPTION
**Pull request by callebtc**

### CODE REVIEW

File: lnbits/wallets/lndrest.py [hunk 1]
```diff
@@ -136,14 +136,25 @@ async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse
...
-            timeout=None,
-        )
+        try:
+            r = await self.client.post(
+                url="/v1/channels/transactions",
+                json={"payment_request": bolt11, "fee_limit": lnrpcFeeLimit},
+                timeout=None,
+            )
+        except Exception as exc:
+            logger.warning(f"LndResWallet pay_invoice HTTPX POST error: {r.text}.")
```
Comment (dni) [inline lnbits/wallets/lndrest.py:146]:
```suggestion
            logger.warning(f"LndRestWallet pay_invoice POST error: {r.text}.")
```

File: lnbits/wallets/lndrest.py [hunk 2]
```diff
@@ -136,14 +136,25 @@ async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse
...
+            return PaymentResponse(None, None, None, None, str(exc))
+
+        if r.is_error:
+            logger.warning(f"LndResWallet pay_invoice request error: {r.text}.")
```
Comment (dni) [inline lnbits/wallets/lndrest.py:150]:
```suggestion
            logger.warning(f"LndRestWallet pay_invoice request error: {r.text}.")
```

File: lnbits/wallets/lndrest.py [hunk 3]
```diff
@@ -136,14 +136,25 @@ async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse
...
+            return PaymentResponse(None, None, None, None, r.text)
+
+        data = r.json()
 
-        if r.is_error or r.json().get("payment_error"):
+        if data.get("payment_error"):
             error_message = r.json().get("payment_error") or r.text
+            logger.warning(f"LndResWallet pay_invoice payment_error: {error_message}.")
```
Comment (dni) [inline lnbits/wallets/lndrest.py:157]:
```suggestion
            logger.warning(f"LndRestWallet pay_invoice payment_error: {error_message}.")
```

**Review by prusnak — DISMISSED** [review]
CI failure: pyright

**Review by dni — APPROVED** [review]

**Review by motorina0 — APPROVED** [review]

**Comment by prusnak** [comment]
Rebased and squashed

**Pull request merged**